moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -109,6 +109,12 @@ void RobotState::allocMemory()
   velocity_ = position_ + robot_model_->getVariableCount();
   // acceleration and effort share the memory (not both can be specified)
   effort_ = acceleration_ = velocity_ + robot_model_->getVariableCount();
+
+  // initialize last row of transformation matrices, which will not be modified by transform updates anymore
+  for (size_t i = 0, end = robot_model_->getJointModelCount() + robot_model_->getLinkModelCount() +
+                           robot_model_->getLinkGeometryCount();
+       i != end; ++i)
+    variable_joint_transforms_[i].makeAffine();
```

Comment:
By name, this does not belong in this function and it results in unnecessary assignments when copying updated RobotStates.

Could you make that a utility function and call it from [here](https://github.com/ros-planning/moveit/blob/06fcc877ce33db0d430082ae2d99cbec01b26625/moveit_core/robot_state/src/robot_state.cpp#L69) and [here](https://github.com/ros-planning/moveit/blob/06fcc877ce33db0d430082ae2d99cbec01b26625/moveit_core/robot_state/src/robot_state.cpp#L143)?

This can also include the duplicated [`all transforms are dirty initially](https://github.com/ros-planning/moveit/blob/06fcc877ce33db0d430082ae2d99cbec01b26625/moveit_core/robot_state/src/robot_state.cpp#L70) code.